tests/src/cgeo/geocaching/utils/UriUtilsTest.java

```diff
@@ -12,27 +12,37 @@
 
     //Example for folder /cgeo
     private static final String DOC_URI_EXAMPLE = "content://com.android.externalstorage.documents/tree/primary%3ADocuments%2Fcgeo/document/primary%3ADocuments%2Fcgeo%2Flogfiles%2Flogcat_2020-12-28_17-22-20-2.txt";
-    //Example for folder /Documents/cgeo
-    private static final String DOC_URI_EXAMPLE_2 = "content://com.android.externalstorage.documents/tree/home%3Acgeo";
-    //Example for folder /Downloads/cgeo
-    private static final String DOC_URI_EXAMPLE_3 = "content://com.android.externalstorage.documents/tree/primary%3ADownload%2Fcgeo";
-
-    //complicated example for a document in /Documents/cgeo/gpx
-    private static final String DOC_URI_COMPLICATED = "content://com.android.externalstorage.documents/tree/home%3Acgeo/document/home%3Acgeo%2Fgpx%2Froute_2021-01-06_17-14-08-4.gpx";
 
     private static final String DOC_URI_EXAMPLE_DECODED = "content://com.android.externalstorage.documents/tree/primary:Documents/cgeo/document/primary:Documents/cgeo/logfiles/logcat_2020-12-28_17-22-20-2.txt";
 
-    private static final String FILE_URI_EXAMPLE = "file:///storage/emulated/0/cgeo/test.txt";
+    //real-world Uri examples and their wanted display
+    //Each test case has entries "Description", "expected display Uri", "raw Uri
+    public static final String[][] URI_DISPLAY_TESTS = new String[][]{
+        new String[]{"Internal /cgeo", "/cgeo", "content://com.android.externalstorage.documents/tree/primary%3Acgeo"},
+        new String[]{"Internal /cgeo/logfiles", "/cgeo/logfiles", "content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Flogfiles"},
+        new String[]{"Internal file in /cgeo/logfiles", "/cgeo/logfiles/log.txt", "content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Flogfiles%2Flog.txt"},
+        new String[]{"SDCARD /cgeo with known volume id", "TEST-SDCARD/cgeo", "content://com.android.externalstorage.documents/tree/TEST-1111-2222%3Acgeo"},
+        new String[]{"SDCARD /Music/cgeo with unknown volume id", "TEST-UNKNOWNID/Music/cgeo", "content://com.android.externalstorage.documents/tree/TEST-UNKNOWNID%3AMusic%2Fcgeo"},
+        new String[]{"SDCARD /cgeo with unknown volume id", "TEST-UNKNOWNID/cgeo", "content://com.android.externalstorage.documents/tree/TEST-UNKNOWNID%3Acgeo"},
```

Comment:
Should we really display "TEST-UNKNOWNID"? Wouldn't `...` be better in this case?

Reply:
This is just like this in the test case. You may see that the same id appears also in the Uri. The test cases tests that whatever is in this place in the Uri (usually a UUID of a volume) is displayed there. And this only happens if c:geo cannot find the volumne name in the volume manager.

Reply:
Will it always be "TEST-UNKNOWNID" literally or is there another way to detect such cases? Then this scenario should be handled. However, I fear that it is not so easy...

Reply:
Again, this is for unit test only. In real life there will for sure not be a volume uuid in such a strange format. The real world uuids I saw had the Format xxxx-xxxx, where each x is a hexadecimal digit. Example: 1AD3-34FE. But of course we can't be sure what can happen out there in the real world.

Also, this is will only be shown to the user if Androids VolumeManager does not know the name for this volume. In the most common case such volume s name seems to be SDCARD.

Reply:
Sorry, now I got it. Thanks for clarification...